pkg/build/util/util.go

```diff
@@ -75,7 +76,30 @@ func IsBuildComplete(build *buildapi.Build) bool {
 }
 
 // BuildNameForConfigVersion returns the name of the version-th build
-// for the config that has the provided name
+// for the config that has the provided name.
 func BuildNameForConfigVersion(name string, version int) string {
 	return fmt.Sprintf("%s-%d", name, version)
 }
+
+// ConfigNameForBuild returns the name of the build config from a
+// build name.
+func ConfigNameForBuild(build *buildapi.Build) string {
+	if build == nil {
+		return ""
+	}
+	return build.Labels[buildapi.DeprecatedBuildConfigLabel]
```

Comment:
you should be checking for buildapi.BuildConfigLabel first, and falling back to buildapi.DeprecatedBuildConfigLabel, as we've discussed elsewhere.

again we really need @php-coder to go scrub all the usage of DeprecatedBuildConfigLabel and BuildConfigLabel to ensure all that code is checking for either label.